src/NJsonSchema/Converters/JsonInheritanceConverter.cs

```diff
@@ -163,19 +163,28 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                 return null;
             }
 
-            var discriminator = jObject.GetValue(_discriminator).Value<string>();
-            var subtype = GetDiscriminatorType(jObject, objectType, discriminator);
-
-            var objectContract = serializer.ContractResolver.ResolveContract(subtype) as JsonObjectContract;
-            if (objectContract == null || objectContract.Properties.All(p => p.PropertyName != _discriminator))
+            Type deserializeType = null;
+            if (_baseType != null && objectType != _baseType)

```

Comment:
This part is quite dangerous I think. What if the objectType is not the baseType but still not the actual desired type (e.g. B in A => B => C)?